File: content/examples/other/leetcode/problems/reverse-bits/example-1/solution.py
def reverse_bits_scan(n):
    n &= 0xFFFFFFFF
    result = 0
    for _ in range(32):
        result = (result << 1) | (n & 1)
        n >>= 1
    return result & 0xFFFFFFFF


if __name__ == "__main__":
    print(reverse_bits_scan(0b00000010100101000001111010011100))

File: content/examples/other/leetcode/problems/reverse-bits/example-1/solution.js
function reverseBitsScan(n) {
  n = n >>> 0;
  let result = 0;
  for (let i = 0; i < 32; i += 1) {
    result = (result << 1) | (n & 1);
    n >>>= 1;
  }
  return result >>> 0;
}

if (require.main === module) {
  console.log(reverseBitsScan(0b00000010100101000001111010011100));
}

module.exports = { reverseBitsScan };
